### DISCUSSION TITLE
R targets error "Error : invalid 'path' argument" when adding git2r  package tar_option_set()

### DISCUSSION THREAD

**Discussion by SimonCoulombe** [ropensci/targets#793] [Help] [answered]
Hi everyone,

I am stumped.  Here is a minimal working example.

This works:

    # _targets.R file
    
    library(targets)
    
    tar_option_set(packages = c("xgboost", "dplyr"))
    
    train_xgboost_model <- function(){
      data(iris)
      dmat <-
        xgboost::xgb.DMatrix(
          data        = iris %>% select(-Species) %>% as.matrix(),
          label       = iris %>% pull(Species)
        )
      xgb_model <- xgb_model <-
        xgboost::xgb.train(
          data = dmat,
          nrounds = 20,
          max_depth = 5,
          eta = 0.2,
          nthread = 5
        )
      return(xgb_model)
    }
    
    
    list(
      tar_target(xgb_iris, train_xgboost_model())
    )

returning:

```
> targets::tar_make()
• start target xgb_iris
• built target xgb_iris
• end pipeline
```

But this doesnt.  I just added "git2r" to the tar_option_set()

    # _targets.R file
    
    library(targets)
    
    tar_option_set(packages = c("xgboost", "dplyr", "git2r"))
    
    train_xgboost_model <- function(){
      data(iris)
      dmat <-
        xgboost::xgb.DMatrix(
          data        = iris %>% select(-Species) %>% as.matrix(),
          label       = iris %>% pull(Species)
        )
      xgb_model <- xgb_model <-
        xgboost::xgb.train(
          data = dmat,
          nrounds = 20,
          max_depth = 5,
          eta = 0.2,
          nthread = 5
        )
      return(xgb_model)
    }
    
    
    list(
      tar_target(xgb_iris, train_xgboost_model())
    )

Here is the error:

    > targets::tar_make()
    • start target xgb_iris
    x error target xgb_iris
    • end pipeline
    Error : invalid 'path' argument
    Error: callr subprocess failed: invalid 'path' argument
    Visit https://books.ropensci.org/targets/debugging.html for debugging advice.
    Run `rlang::last_error()` to see where the error occurred.

Here is my sessioninfo::session_info()

 ```

> sessioninfo::session_info()
─ Session info  ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 hash: bug, stop sign, sign of the horns: light skin tone

 setting  value
 version  R version 3.6.3 (2020-02-29)
 os       OpenShift Enterprise
 system   x86_64, linux-gnu
 ui       RStudio
 language (EN)
 collate  en_CA.UTF-8
 ctype    en_CA.UTF-8
 tz       America/Toronto
 date     2022-03-02
 rstudio  1.4.1717.3 Juliet Rose (server)
 pandoc   NA

─ Packages ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 package     * version date (UTC) lib source
 backports     1.4.0   2021-11-23 [1] RSPM (R 3.6.3)
 base64url     1.4     2018-05-14 [1] RSPM (R 3.6.1)
 callr         3.7.0   2021-04-20 [1] RSPM (R 3.6.3)
 cli           3.1.0   2021-10-27 [1] RSPM (R 3.6.3)
 codetools     0.2-16  2018-12-24 [2] CRAN (R 3.6.3)
 crayon        1.4.2   2021-10-29 [1] RSPM (R 3.6.3)
 data.table    1.14.2  2021-09-27 [1] RSPM (R 3.6.3)
 digest        0.6.28  2021-09-23 [1] RSPM (R 3.6.3)
 ellipsis      0.3.2   2021-04-29 [1] RSPM (R 3.6.3)
 fansi         0.5.0   2021-05-25 [1] RSPM (R 3.6.3)
 glue          1.5.0   2021-11-07 [1] RSPM (R 3.6.3)
 igraph        1.2.11  2022-01-04 [1] RSPM (R 3.6.3)
 jsonlite      1.7.2   2020-12-09 [1] RSPM (R 3.6.3)
 knitr         1.36    2021-09-29 [1] RSPM (R 3.6.3)
 lattice       0.20-45 2021-09-22 [1] RSPM (R 3.6.3)
 lifecycle     1.0.1   2021-09-24 [1] RSPM (R 3.6.3)
 magrittr      2.0.1   2020-11-17 [1] RSPM (R 3.6.3)
 Matrix        1.2-18  2019-11-27 [2] CRAN (R 3.6.3)
 pillar        1.6.4   2021-10-18 [1] RSPM (R 3.6.3)
 pkgconfig     2.0.3   2019-09-22 [1] CRAN (R 3.6.1)
 processx      3.5.2   2021-04-30 [1] RSPM (R 3.6.3)
 ps            1.6.0   2021-02-28 [1] RSPM (R 3.6.3)
 purrr         0.3.4   2020-04-17 [1] CRAN (R 3.6.1)
 R6            2.5.1   2021-08-19 [1] RSPM (R 3.6.3)
 rlang         0.4.12  2021-10-18 [1] RSPM (R 3.6.3)
 rstudioapi    0.13    2020-11-12 [1] RSPM (R 3.6.3)
 sessioninfo   1.2.1   2021-11-02 [1] RSPM (R 3.6.3)
 targets       0.10.0  2022-01-07 [1] RSPM (R 3.6.3)
 tibble        3.1.6   2021-11-07 [1] RSPM (R 3.6.3)
 tidyselect    1.1.1   2021-04-30 [1] RSPM (R 3.6.3)
 utf8          1.2.2   2021-07-24 [1] RSPM (R 3.6.3)
 vctrs         0.3.8   2021-04-29 [1] RSPM (R 3.6.3)
 withr         2.4.2   2021-04-18 [1] RSPM (R 3.6.3)
 xfun          0.28    2021-11-04 [1] RSPM (R 3.6.3)
 xgboost       1.5.0.2 2021-11-21 [1] RSPM (R 3.6.3)
 yaml          2.2.1   2020-02-01 [1] RSPM (R 3.6.2)

 [1] /home/eg66451/R/x86_64-pc-linux-gnu-library/3.6
 [2] /opt/R/3.6.3/lib/R/library

```

**Comment by wlandau** [comment, marked as answer]
`git2r::pull()` is masking `dplyr::pull()` in your code. If you replace `pull(Species)` with `dplyr::pull(Species)`, it will work.

**Reply by SimonCoulombe** [reply]
ooh. my bad.  thanks!

I'm trying to build a reproducible example for another fun issue.  I get  "quitting from line XXX-XXX (report.Rmd)" , no error, just an infinite loop of   / \  / \  / \  spinning around.
